### PR TITLE
fix: refresh reference lines on chart update

### PR DESCRIPTION
**Pull request by rephus**
Closes: https://github.com/lightdash/lightdash/issues/4141
Closes: #4133

### Description:
This replaces https://github.com/lightdash/lightdash/pull/4154 

I moved the series update to `useCartesianChartConfig` so we can apply markline config when series change
### Acceptance criteria

If I have a reference line, I'd like to keep my reference line when:

- [x] A group changes

[Screencast from 12-01-23 12:00:41.webm](https://user-images.githubusercontent.com/1983672/212051945-0f5ec145-1295-4ef3-b033-6a41072e67ab.webm)

image

- [x] A chart is flipped

[Screencast from 12-01-23 11:29:00.webm](https://user-images.githubusercontent.com/1983672/212043144-94f6d7ec-55dd-4a30-95b5-b3307fad752a.webm)

### CODE REVIEW

File: packages/frontend/src/components/common/ReferenceLine.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    xField: string | undefined,
+    flipAxes: boolean,
+    fieldId: string,
+): string => {
+    const reverseFlippedAxis = (defaultAxis: string) => {
+        if (flipAxes) return defaultAxis === 'xAxis' ? 'yAxis' : 'xAxis';
+        return defaultAxis;
+    };
+
+    return reverseFlippedAxis(xField === fieldId ? 'xAxis' : 'yAxis');
```
Comment (ZeRego) [inline packages/frontend/src/components/common/ReferenceLine.tsx:18]: Found this logic hard to follow with the internal function and the logic around the string and not the boolean.
Let me know what you think about this:
```
const isDefaultXAxis = xField === fieldId;
if(flipAxes) {
   return isDefaultXAxis ? 'yAxis' : 'xAxis'
}
return isDefaultXAxis ? 'xAxis' : 'yAxis'
```

**Comment by ZeRego** [comment]
After saving, the reference line is still in the config panel but doesn't show in the chart.
Screenshot 2023-01-13 at 10 41 32
Chart:
https://lightdash-pr-4155.onrender.com/projects/90614fc7-3ad5-4c6a-9113-e99681f35fb4/saved/aefa566a-86b7-430e-a3ae-32dea21b5128/edit
Video:
https://www.loom.com/share/6482260d14ba4736af77ddba77b5139a

**Comment by rephus** [comment]
> After saving, the reference line is still in the config panel but doesn't show in the chart. ![Screenshot 2023-01-13 at 10 41 

Fixed

**Review by ZeRego — APPROVED** [review]
